src/seq_mv/vector_device.c

```diff
@@ -339,6 +339,32 @@ hypre_SeqVectorSumEltsDevice( hypre_Vector *vector )
    return sum;
 }
 
+/*--------------------------------------------------------------------------
+ * hypre_SeqVectorStridedCopyDevice
+ *--------------------------------------------------------------------------*/
+
+HYPRE_Int
+hypre_SeqVectorStridedCopyDevice( hypre_Vector  *vector,
+                                  HYPRE_Int      stride,
+                                  HYPRE_Int      size,
+                                  HYPRE_Complex *data)
+{
+   HYPRE_Complex  *v_data = hypre_VectorData(vector);
+
+#if defined(HYPRE_USING_CUDA) || defined(HYPRE_USING_HIP)
+   auto counting_iter = thrust::make_counting_iterator(0);
+
+   HYPRE_THRUST_CALL( transform_if, counting_iter, counting_iter + size, v_data, v_data,
```

Comment:
This call has the complexity of O(size). There should be a more efficient implementation of O(size/stride), I assume.

Reply:
Yes, done in https://github.com/hypre-space/hypre/pull/1054/commits/b507d9cc0e54939e4104772eae57ff1225bd23d8

I've also generalized the new code to accept different strides for the source and destination arrays